packages/tldraw/src/lib/ui/context/actions.tsx

```diff
@@ -1565,6 +1578,47 @@ export function ActionsProvider({ overrides, children }: ActionsProviderProps) {
 					}
 				},
 			},
+			{
+				id: 'download-original',
+				label: 'action.download-original',
+				readonlyOk: true,
+				onSelect: async (source) => {
+					const selectedShapes = editor.getSelectedShapes()
+					if (selectedShapes.length === 0) return
+
+					const mediaShapes = selectedShapes.filter((s): s is TLImageShape | TLVideoShape =>
+						supportsDownloadingOriginal(s, editor)
+					)
+
+					// We only want to download originals if we only have media shapes selected
+					if (mediaShapes.length !== selectedShapes.length) return
+
+					for (const mediaShape of mediaShapes) {
+						const asset = editor.getAsset(mediaShape.props.assetId!)
+						if (!asset || !asset.props.src) continue
+
+						const url = await editor.resolveAssetUrl(asset.id, { shouldResolveToOriginal: true })
+						if (!url) return
+
+						const link = document.createElement('a')
+						link.href = url
+
+						if (
+							(asset.type === 'video' || asset.type === 'image') &&
+							!asset.props.src.startsWith('asset:')
+						) {
+							link.download = asset.props.name
```

Comment:
hm, i wonder in the case of multiple downloads if the browser throws a "looks like the website is trying to do many downloads at once". 

what do you think about using `archiver` to bundle the files together?
https://www.npmjs.com/package/archiver
my understanding of `archiver` is it can output a zip file without compression applied (we don't need compression, just a bundler type file format). plus it seems that npm package is pretty lightweight.

Reply:
Yeah my worry is that folks will miss the prompt and then wonder what's going on.
When you mention the backup use-case - that does sound compelling actually!

Reply:
Let's talk archiving as a follow up